src/components/pages/public/AllMoviesPage.vue

```diff
@@ -25,22 +34,44 @@ import BreadcrumbNav from "@/components/BreadcrumbNav.vue";
 import BaseInput from "@/components/base/BaseInput.vue";
 import MovieSelect from "@/components/base/MovieSelect.vue";
 import MovieCard from "@/components/MovieCard.vue";
+import { mapGetters } from "vuex";
 
 export default {
   name: "AllMoviesPage",
   components: { BreadcrumbNav, BaseInput, MovieSelect, MovieCard },
   data() {
     return {
       query: "",
+      selectedOption: "",
     };
   },
   computed: {
-    movies() {
-      return this.$store.state.movies;
+    ...mapGetters(["movies", "genres"]),
+    selectOptions() {
+      return ["All categories", ...this.genres];
     },
+    filteredMovies() {
+      const searchedMovies = this.movies.filter((movie) => {
+        if (!this.query) {
+          return true;
+        } else {
+          return movie.title.toUpperCase().includes(this.query.toUpperCase());
+        }
+      });
```

Comment:
This part could be simplified and use ternary operator:
`this.query ? this.movies.filer((movie) => movie.title.toUpperCase().includes(this.query.toUpperCase())) : this.movies`.
Also this part could be extracted another computed property so we would know that these are movies filtered by query.